crates/subspace-farmer/src/single_disk_plot.rs

```diff
@@ -618,6 +621,7 @@ impl SingleDiskPlot {
 
                             sector_metadata
                                 .copy_from_slice(&SectorMetadata { expires_at }.encode());
+                            let mut metadata_header = metadata_header.lock();
                             metadata_header.sector_count += 1;
                             metadata_header_mmap
                                 .copy_from_slice(metadata_header.encode().as_slice());
```

Comment:
This is essentially a database, we certainly don't want to keep all changes in memory and potentially miss write to disk due to power loss or other reasons.

Reply:
But we won't miss. Let me sketch it here:
```rust
#[derive(Deref)]
struct MmappedStruct<T> {
    #[deref]
    cached: T,
    mmap: MmapMut,
};

impl<T: Decode + Encode> MmappedStruct<T> {
   fn new(value: T, path: PathBuf) -> io::Result<Self>;
   fn write(&'_ mut self) -> WriteToDiskOnDrop<'_, T>
}

#[derive(DerefMut)
struct WriteToDisk<'a, T> {
    #[deref_mut]
    value: &mut T,
    mmap: &mut MmapMut
}

// Write to disk
impl<T: Decode + Encode> Drop for WriteToDisk<'_, T> {}
```

Reply:
You naively assume application will always be gracefully shut down and not killed/losing power accidentally, etc.
Also this will not help if we want to farm already plotted sectors that are stuck in the cache.
What benefit are you trying to extract with proposed change?

Reply:
I mean, this will be identical in terms of semantics (both will be saved on disk), but api will differ:
```rust
let mut metadata_header = metadata_header.lock();
metadata_header.sector_count += 1;
metadata_header_mmap
    .copy_from_slice(metadata_header.encode().as_slice());
```
Will become this:
```rust
mmapped_metadata_header.write().sector_count += 1;
// Here on drop of `WriteToDiskOnDrop` we will encode new version of struct and write to mmaped buffer
```